### ISSUE TITLE
Error position in function expectDigit

### ISSUE THREAD

**Issue by wakenmeng** [josdejong/jsonrepair#98]
Hi, there

The `position` value of `JSONRepairError` in function [expectDigit](https://github.com/josdejong/jsonrepair/blob/7cd44c98d39a0a702b69b38884157ea7d9ff4064/src/jsonrepair.ts#L609C21-L609C21) should be `i`, instead of `2`, right?

```
  function expectDigit(start: number) {
    if (!isDigit(text.charCodeAt(i))) {
      const numSoFar = text.slice(start, i)
      throw new JSONRepairError(`Invalid number '${numSoFar}', expecting a digit ${got()}`, 2)
    }
  }
```

**Comment by josdejong** [comment]
O wow, how did that `2` get in the code there 😂 . Thanks for reporting Waken.

**Comment by wakenmeng** [comment]
Hi, @josdejong You're welcome. This project has been very helpful to me. I really enjoy it and appreciate your work very much. I wrote a Go implementation [link](https://github.com/wakenmeng/jsonrepair) of it, with all the test cases included. If you are interested, feel free to take a look. Or if you have any feedback, please don't hesitate to contact me.

**Comment by josdejong** [comment]
Nice! Thanks for sharing.

**Comment by josdejong** [comment]
This has been fixed now in `v3.2.1`
